src/ucp/rndv/rndv.c

```diff
@@ -544,33 +547,51 @@ ucp_rndv_progress_rma_zcopy_common(ucp_request_t *req, ucp_lane_index_t lane,
     }
 }
 
+static UCS_F_ALWAYS_INLINE ucp_request_t*
+ucp_rndv_rma_get_zcopy_req(ucp_request_t *req);
+
 UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_progress_rma_get_zcopy, (self),
                  uct_pending_req_t *self)
 {
-    ucp_request_t *rndv_req = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_request_t *rndv_req;
     uct_rkey_t uct_rkey;
 
+    rndv_req = ucp_rndv_rma_get_zcopy_req(req);
+
     /* Figure out which lane to use for get operation */
-    rndv_req->send.lane =
-        ucp_rndv_zcopy_get_lane(rndv_req, &uct_rkey,
-                                UCP_REQUEST_SEND_PROTO_RNDV_GET);
-    if (rndv_req->send.lane != UCP_NULL_LANE) {
+    req->send.lane =
+            ucp_rndv_zcopy_get_lane(req, &uct_rkey,
+                                    UCP_REQUEST_SEND_PROTO_RNDV_GET);
+    if (req->send.lane != UCP_NULL_LANE) {
```

Comment:
don't we check the same thing in line 821?

Reply:
removed from here